src/sentry/static/sentry/app/views/eventsV2/eventView.tsx

```diff
@@ -27,32 +34,57 @@ type Field = {
   // width: number;
 };
 
-const decodeFields = (location: Location): Array<Field> => {
-  const {query} = location;
-
-  if (!query || !query.field) {
-    return [];
-  }
-
-  const fields: string[] = isString(query.field) ? [query.field] : query.field;
-  const aliases: string[] = Array.isArray(query.alias)
-    ? query.alias
-    : isString(query.alias)
-    ? [query.alias]
-    : [];
+const decodeFields = (fields: string[], aliases: string[]): Field[] => {
+  // invariant: fields and aliases should have equal array lengths
 
   const parsed: Field[] = [];
-  fields.forEach((field, i) => {
+  fields.forEach((field, index) => {
     let title = field;
-    if (aliases[i]) {
-      title = aliases[i];
+    if (aliases[index]) {
+      title = aliases[index];
     }
     parsed.push({field, title});
   });
 
   return parsed;
 };
 
+type DiscoverState = {
+  fields: string[];
```

Comment:
Should the name of the query view go into the state as well? What about the datetime, query text and project selections? Those are all part of the saved query too.

Reply:
I can put some query strings into the state to be compressed; but not all. 

The query strings here are set by the component that controls the project selection, env selection, and the time span: https://github.com/getsentry/sentry/blob/05a1de9d32fefa5676c8065edd73f044ce9aa9ac/src/sentry/static/sentry/app/views/eventsV2/eventView.tsx#L366-L387

Reply:
I forgot about global selection state needing to be managed elsewhere.